packages/playwright-core/src/utils/utils.ts

```diff
@@ -405,6 +405,7 @@ export function arrayToObject(array?: NameValue[]): { [key: string]: string } |
 export async function calculateFileSha1(filename: string): Promise<string> {
   const hashStream = new HashStream();
   const stream = fs.createReadStream(filename);
+  stream.once('error', err => hashStream.emit('error', err));
```

Comment:
Node convention for streams is to dispatch an error event upon failures. And if error handler is not attached, event target will generate unhandled exception. The right way to handle it is to register 'error' event handler forever (on, not once), and upon error, in this case, reject the resulting promise. You can use Promise.race for it here, no need to dispatch it on the foreign hash stream object.

Reply:
Updated in https://github.com/microsoft/playwright/commit/83e988b9b0574c541adce3e3a00f9dd0a32297da.

Let me know if I created a race with the error listener. 😀 I think I'm fine since there are no awaits between where we create the readstream and I register err listener promise.

@pavelfeldman For my own education, when does Node actually go to open the file? Is it as soon as we give over the thread while waiting for the settling of thee `race` race promise?